src/server.ts

```diff
@@ -699,6 +724,75 @@ export class Server {
       offset,
     );
   }
+
+  /**
+   * Check if any of the destination accounts requires a memo.
+   *
+   * This function implements a memo required check as defined in
+   * [SEP0029](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md).
+   * It will load each account which is the destination and check if it has the
+   * data field `config.memo_required` set to `"MQ=="`.
+   *
+   * Each account is checked sequentially instead of loading multiple accounts
+   * at the same time from Horizon.
+   *
+   * @see
+   * [SEP0029](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md)
+   * @param {Transaction} transaction - The transaction to check.
+   * @returns {Promise<void, Error>} - If any of the destination account
+   * requires a memo, the promise will throw {@link AccountRequiresMemoError}.
+   * @throws  {AccountRequiresMemoError}
+   */
+  public async checkMemoRequired(transaction: Transaction): Promise<void> {
+    if (transaction.memo.type !== "none") {
+      return;
+    }
+
+    const destinations = new Set<string>();
+
+    for (let i = 0; i < transaction.operations.length; i++) {
+      const operation = transaction.operations[i];
+
+      switch (operation.type) {
+        case "payment":
+        case "pathPaymentStrictReceive":
+        case "pathPaymentStrictSend":
+        case "accountMerge":
+          break;
+        default:
+          continue;
+      }
+      const destination = operation.destination;
+      if (destinations.has(destination)) {
+        continue;
+      }
```

Comment:
I think these lines of code are unnecessary. The `destinations`, if it is a `Set`, should be doing this check.
```suggestion
```

Reply:
@leighmcculloch what do you mean? this is to skip the check if it was already checked.

Reply:
Ah right. You're right! Please disregard this comment.